scripts/base/frameworks/storage/main.zeek

```diff
@@ -0,0 +1,170 @@
+##! The storage framework provides a way to store long-term data to disk.
+
+@load base/bif/storage.bif
+
+module Storage;
+
+export {
+	## Record for passing arguments to ``put``.
+	type PutArgs: record {
+		backend: opaque of Storage::BackendHandle;
+
+		# The script-level type of keys stored in the backend. Used for
+		# validation of keys passed to other framework methods.
+		key: any;
+
+		# The script-level type of keys stored in the backend. Used for
+		# validation of values passed to ``put`` as well for type
+		# conversions for return values from ``get``.
+		value: any;
+
+		# Indicates whether this value should overwrite an existing entry
+		# for the key.
+		overwrite: bool &default=F;
+
+		# An interval of time until the entry is automatically removed from the
+		# backend.
+		expire_time: interval &default=0sec;
+
+		# Indicates whether this operation should happen
+		# asynchronously. If this is true, the call to put must happen
+		# as part of a when statement. This flag is overridden when
+		# reading pcaps, since time won't move forward the same as when
+		# caputring live traffic.
+		async_mode: bool &default=T;
+	};
+
+	## Options record for the built-in SQLite backend.
+	type SqliteOptions: record {
+		## Path to the database file on disk
```

Comment:
Note, `database_path=":memory:"` can be used for in-memory dbs.

Reply:
Added a note in the comment about it.